client/serviceinterface_create.go

```diff
@@ -16,6 +18,59 @@ func (cli *VanClient) ServiceInterfaceCreate(ctx context.Context, service *types
 		if err != nil {
 			return err
 		}
+
+		if len(service.TlsCredentials) > 0 {
+
+			configmap, err := cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Get(types.TransportConfigMapName, metav1.GetOptions{})
+
+			if err != nil {
+				return err
+			}
+
+			serviceCredential := types.Credential{
+				CA:          types.ServiceCaSecret,
+				Name:        service.TlsCredentials,
+				Subject:     service.Address,
+				Hosts:       []string{service.Address},
+				ConnectJson: false,
+				Post:        false,
+			}
+
+			ownerReference := metav1.OwnerReference{
+				APIVersion: "v1",
+				Kind:       "ConfigMap",
+				Name:       configmap.Name,
+				UID:        configmap.UID,
+			}
+			serviceSecret, err := kube.NewSecret(serviceCredential, &ownerReference, cli.Namespace, cli.KubeClient)
+			if err != nil {
+				return err
+			}
+
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+				err := qdr.AddSslProfile(serviceSecret.Name, cli.Namespace, cli.KubeClient)
+				if err != nil {
+					return err
```

Comment:
@nluaces Should this be part of the retry logic? I am wondering if just the deployment
update fails, then the `sslProfile` will be already defined, in case of a possible retry attempt.

Reply:
Agree, it is changed now. Thanks!